### PR TITLE
Add license header checks

### PR DESCRIPTION
**Pull request by slfuchs**
The goal is to ensure that the license header of files in the repo is/remains compliant with the license header as defined in the file `utilities/license-header/template-license-header`. The year(s) within the copyright statement is set to the year a file is added to the repo (e.g., 2021), or to the time span a file is created and modified last (e.g. 2021-2025).

The existing `Indent` workflow is renamed to `Code Checks` and covers the jobs `clang-format` and `license-header`. Note that the job `license-header` is executed
* on `push` and `pull_request` to master for changed files only, i.e., files returned by the command `git diff --name-only --diff-filter=ACMR origin/master...HEAD` (where deleted files are ignored).
* on a weekly schedule for all files in the repository.

### CODE REVIEW

File: .github/workflows/codechecks.yml [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+          clangFormatVersion: 11
+          inplace: true
+
+  license-header:
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
```
Comment (slfuchs) [inline .github/workflows/codechecks.yml:32]: This ensures that the complete repository is checked out, such that `git diff` can be executed.

**Comment by slfuchs** [comment]
@ppraegla This PR is ready for review.

**Review by ppraegla — APPROVED** [review]
Two minor remarks or reminders:
1. We need a way to exclude files from the license check. E.g. doc/sphinx/conv.py. Currently, it is excluded because we don't check python files since we don't have any.
2. I think at some point we need to come up with a way to allow a shebang (e.g. `#!/usr/bin/env python`) before the license header. AFAIK the shebang must be the first line of the file which conflicts with our license header that is currently starting at the first line.

Though, these two points can be tackled once they appear. Thanks for the PR.

**Comment by slfuchs** [comment]
Yes, I already had both points in mind.

> 1. We need a way to exclude files from the license check. E.g. doc/sphinx/conv.py. Currently, it is excluded because we don't check python files since we don't have any.

I also think that we can skip this point until we actually need it.

> 2. I think at some point we need to come up with a way to allow a shebang (e.g. `#!/usr/bin/env python`) before the license header. AFAIK the shebang must be the first line of the file which conflicts with our license header that is currently starting at the first line.

I just tackled that, so now we also can check files having a shebang in the first line, e.g., the scripts added with this pull request.

**Comment by slfuchs** [comment]
Besides that I added a short description to the Sphinx Documentation.

**Comment by slfuchs** [comment]
@ppraegla Please take a look again at the recent changes (see comments above) and feel free to merge if you agree.

**Review by ppraegla — APPROVED** [review]

**Pull request merged**